src/main/kotlin/day16/util/CommonUtils.kt

```
@@ -0,0 +1,23 @@
+    // 날짜 변환 메소드
+    fun formattedTime(timeInMillis: Long): String {
+        // Duration은 나노초 단위까지 지원
+        val duration = Duration.ofMillis(timeInMillis)
+
+        val hours = duration.toHours()
+        val minutes = duration.toMinutes() % 60
+        val seconds = duration.seconds % 60
+        val milliseconds = duration.toMillis() % 1000
+        val micros = (timeInMillis % 1000) * 1000
+
+        return String.format("%02d:%02d:%02d.%03d%03d", hours, minutes, seconds, milliseconds, micros)
```

detekt (1.23.7)

[warning] String.format("%02d:%02d:%02d.%03d%03d", hours, minutes, seconds, milliseconds, micros) uses implicitly default locale for string formatting.

(detekt.potential-bugs.ImplicitDefaultLocale)